src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -833,6 +840,12 @@ public void RemoveNode(NodeModel model)
 
         protected virtual void DisposeNode(NodeModel model)
         {
+            var functionNode = model as Function;
+            if (functionNode != null)
+            {
+                functionNode.Controller.SyncWithDefinitionStart -= OnSyncWithDefintionStart;
+                functionNode.Controller.SyncWithDefinitionEnd -= OnSyncWithDefinitionEnd;
+            }
```

Comment:
Yep, pretty much addressing the [small concern I had](https://github.com/DynamoDS/Dynamo/pull/5518#discussion_r42077927) with the previous PR. Looks good!